### PR TITLE
Add documentation on bot_check_once

### PR DESCRIPTION
**Pull request by JJgar2725**
### Summary

This pull request implements some information on the bot_check_once function. Let me know if any revisions to my edits would be necessary for the documentation on this topic.

Issue: #4193 

### Checklist

- [ ] If code changes were made then they have been tested.
    - [ ] I have updated the documentation to reflect the changes.
- [ ] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [x] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: discord/ext/commands/cog.py [hunk 1]
```diff
@@ -296,6 +296,10 @@ def bot_check_once(self, ctx):
         """A special method that registers as a :meth:`.Bot.check_once`
         check.
 
+        Note that when using this function, context sent from a group command
+        may only parse the parent command and not the subcommands due to it
+        being invoked once per :meth:`.Command.invoke` call.
+
```
Comment (Rapptz) [inline discord/ext/commands/cog.py:302]: In rst we use something called note blocks that look like...

```rst
.. note::

    text goes here, note that it's indented by 4 spaces
```

File: discord/ext/commands/cog.py [hunk 2]
```diff
@@ -296,6 +296,10 @@ def bot_check_once(self, ctx):
         """A special method that registers as a :meth:`.Bot.check_once`
         check.
 
+        Note that when using this function, context sent from a group command
+        may only parse the parent command and not the subcommands due to it
+        being invoked once per :meth:`.Command.invoke` call.
```
Comment (Rapptz) [inline discord/ext/commands/cog.py:301]:
```suggestion
        being invoked once per :meth:`.Bot.invoke` call.
```

File: discord/ext/commands/cog.py [hunk 3]
```diff
@@ -296,6 +296,10 @@ def bot_check_once(self, ctx):
         """A special method that registers as a :meth:`.Bot.check_once`
         check.
 
+        Note that when using this function, context sent from a group command
```
Comment (Rapptz) [inline discord/ext/commands/cog.py:299]:
```suggestion
       When using this function the :class:`.Context` sent to a group subcommand
```

**Comment by JJgar2725** [comment]
Okay, nice! I'll make the changes in a bit. 👍

**Comment by Rapptz** [comment]
Looks good, thanks.

**Pull request merged**